The operations subsystem is a singular entity

sykesm opened the pull request:
Update the error message to use the singular "subsystem" instead of "subsystems".

Relates to #797

ghaskins commented:
@srderson @jeffgarratt @jyellick @muralisrini

jyellick commented:
I can confirm that combined with pull request https://github.com/openblockchain/obc-dev-env/pull/49

All go tests pass.
All behave tests pass.

Would love to hear confirmation from @jeffgarratt @srderson or @muralisrini, but I'm comfortable merging this after the obc-dev-env pull request.

bcbrock commented:
Who, or what organization is going to provide and maintain the Docker openblockchain/baseimage image? If IBM, have the legal issues been resolved? Is this person/organization also willing to sponsor base images for POWER and Z?

ghaskins commented:
@bcbrock As we discussed last week f2f, the current proposal is as follows:

Regarding ownership: 

1) currently both IBM and myself (LSEG) have administrative rights to the repositories and may push images.  This arrangement is largely a function of it was myself who proposed/created the repositories initially and then subsequently granted IBM ownership/administrative rights.
2) I consider IBM/OBC community authoritative on the repository and only update the repository in strict accordance with agreement/sign-off with the OBC project gatekeepers.
3) I only retain administrative rights as a convenience to the OBC community for as long as the OBC governance wishes me to remain in this role.

Regarding legal issues:

IANAL, but the current repositories only contain upstream FOSS using standard practices accepted within the ecosystem (Vagrant/Docker) so I can't imagine there are many issues to resolve w.r.t. the legality of the repositories themselves.  I imagine the bigger issue is from the OBC community itself w.r.t. who or what is perceived to be in charge of the repository carrying an openblockchain trademark type thing.

Regarding this, I personally don't see even the repository governance as a big deal:  if someone wants to they may simply fork the OBC code and point it at an alternate location.  I am open to suggestion for alternate strategies.

Regarding POWER/Z:

I don't personally have a problem with this especially if we can automate their generation with the tooling. As mentioned above, I would look to approval from the OBC community before taking such a step.

ghaskins commented:
Superseded by PR #922

sykesm committed:
The operations subsystem is a singular entity

Update the error message to use the singular "subsystem" instead of
"subsystems".

Signed-off-by: Matthew Sykes <sykesmat@us.ibm.com>
Change-Id: Icd02d593cb39a69a551f58f9ceed644d9afea0ca

yacovm reviewed (approved):
The singularity is nigh!
